File: java/brasil-burger-java/src/main/java/com/brasilburger/models/Burger.java
package com.brasilburger.models;

import java.math.BigDecimal;
import java.sql.Timestamp;

public class Burger {

    private int id;
    private String libelle;
    private String description;
    private BigDecimal prix;
    private String imageUrl;
    private boolean isArchived;
    private Integer categorieId;
    private String categorieNom;
    private Timestamp createdAt;

    public Burger() {
    }

    public Burger(String libelle, String description, BigDecimal prix, String imageUrl, Integer categorieId) {
        this.libelle = libelle;
        this.description = description;
        this.prix = prix;
        this.imageUrl = imageUrl;
        this.categorieId = categorieId;
        this.isArchived = false;
    }

    // Getters et Setters
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getLibelle() {
        return libelle;
    }

    public void setLibelle(String libelle) {
        this.libelle = libelle;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public BigDecimal getPrix() {
        return prix;
    }

    public void setPrix(BigDecimal prix) {
        this.prix = prix;
    }

    public String getImageUrl() {
        return imageUrl;
    }

    public void setImageUrl(String imageUrl) {
        this.imageUrl = imageUrl;
    }

    public boolean isArchived() {
        return isArchived;
    }

    public void setArchived(boolean archived) {
        isArchived = archived;
    }

    public Integer getCategorieId() {
        return categorieId;
    }

    public void setCategorieId(Integer categorieId) {
        this.categorieId = categorieId;
    }

    public String getCategorieNom() {
        return categorieNom;
    }

    public void setCategorieNom(String categorieNom) {
        this.categorieNom = categorieNom;
    }

    public Timestamp getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Timestamp createdAt) {
        this.createdAt = createdAt;
    }

    @Override
    public String toString() {
        return String.format("Burger #%d - %s | Prix: %s FCFA | Catégorie: %s | %s",
                id, libelle, prix,
                categorieNom != null ? categorieNom : "N/A",
                isArchived ? "ARCHIVÉ" : "ACTIF");
    }

    public String toDetailString() {
        StringBuilder sb = new StringBuilder();
        sb.append("\n╔════════════════════════════════════════════════════════╗\n");
        sb.append(String.format("  ID:          %d\n", id));
        sb.append(String.format("  Libellé:     %s\n", libelle));
        sb.append(String.format("  Description: %s\n", description != null ? description : "N/A"));
        sb.append(String.format("  Prix:        %s FCFA\n", prix));
        sb.append(String.format("  Catégorie:   %s\n", categorieNom != null ? categorieNom : "N/A"));
        sb.append(String.format("  Image URL:   %s\n", imageUrl != null ? imageUrl : "N/A"));
        sb.append(String.format("  Statut:      %s\n", isArchived ? "ARCHIVÉ" : "ACTIF"));
        sb.append("╚════════════════════════════════════════════════════════╝\n");
        return sb.toString();
    }
}

File: BrasilBurgerGestionnaire/src/Entity/Burger.php
<?php

namespace App\Entity;

use App\Repository\BurgerRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: BurgerRepository::class)]
#[ORM\Table(name: 'burgers')]
class Burger
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 150)]
    private ?string $libelle = null;

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $description = null;

    #[ORM\Column(type: 'decimal', precision: 10, scale: 2)]
    private ?string $prix = null;

    #[ORM\Column(name: 'imageurl', length: 500, nullable: true)]
    private ?string $imageUrl = null;

    #[ORM\Column(name: 'isarchived', type: 'boolean')]
    private ?bool $isArchived = false;

    #[ORM\Column(name: 'categorieid', nullable: true)]
    private ?int $categorieId = null;

    #[ORM\ManyToOne(targetEntity: BurgerCategorie::class, inversedBy: 'burgers')]
    #[ORM\JoinColumn(name: 'categorieid', referencedColumnName: 'id', nullable: true)]
    private ?BurgerCategorie $categorie = null;

    #[ORM\Column(name: 'createdat', type: 'datetime')]
    private ?\DateTimeInterface $createdAt = null;

    public function __construct()
    {
        $this->createdAt = new \DateTime();
        $this->isArchived = false;
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getLibelle(): ?string
    {
        return $this->libelle;
    }

    public function setLibelle(string $libelle): static
    {
        $this->libelle = $libelle;
        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(?string $description): static
    {
        $this->description = $description;
        return $this;
    }

    public function getPrix(): ?string
    {
        return $this->prix;
    }

    public function setPrix(string $prix): static
    {
        $this->prix = $prix;
        return $this;
    }

    public function getImageUrl(): ?string
    {
        return $this->imageUrl;
    }

    public function setImageUrl(?string $imageUrl): static
    {
        $this->imageUrl = $imageUrl;
        return $this;
    }

    public function isArchived(): ?bool
    {
        return $this->isArchived;
    }

    public function setIsArchived(bool $isArchived): static
    {
        $this->isArchived = $isArchived;
        return $this;
    }

    public function getCategorieId(): ?int
    {
        return $this->categorieId;
    }

    public function getCategorie(): ?BurgerCategorie
    {
        return $this->categorie;
    }

    public function setCategorie(?BurgerCategorie $categorie): static
    {
        $this->categorie = $categorie;
        if ($categorie !== null) {
            $this->categorieId = $categorie->getId();
        }
        return $this;
    }

    public function getCreatedAt(): ?\DateTimeInterface
    {
        return $this->createdAt;
    }

    public function setCreatedAt(\DateTimeInterface $createdAt): static
    {
        $this->createdAt = $createdAt;
        return $this;
    }
}
